### PR TITLE
completes work for task 2

### PR DESCRIPTION
**Pull request by JonathanCrewe**
Is there some way I should have automated creating the databases? 
I just ran the setup script in psql.

### CODE REVIEW

**Comment by JonathanCrewe** [comment]
Merge task 2.

File: app.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const express = require('express')
+const {getTopics } = require('./controllers/topics.controller.js')
+
+
+// Set up. 
+const app = express()
+app.use(express.json())
```
Comment (nc-christian-dutton) [inline app.js:7]: Going to be picky, do you need this yet?

File: app.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+app.use((req, res)=>{
+    res.status(404).send({msg: 'Not Found'})
+})
+
+app.use((err, req, res, next)=>{
+    if (err.status && err.msg) {
+        res.status(err.status).send({msg: err.msg})
+    } else next(err)
+})
+
```
Comment (nc-christian-dutton) [inline app.js:23]: You should add an error handler that can deal with 500's here (don't need to test it though)

File: app.js (hunk 1 above)
Comment (JonathanCrewe) [reply app.js:7]: No, but I'm pretty sure I'm going to.

**Comment by nc-christian-dutton** [comment]
No you don't need to automate creating the db's, you could add the instruction into your readme on how to set them up. 

Looking good overall, left some small suggestions 👍

File: app.js (hunk 2 above)
Comment (JonathanCrewe) [reply app.js:23]: Does express cover this? Is it considered good practice to do it yourself anyway?

**Comment by JonathanCrewe** [comment]
Thanks for the review/feeback!

**Pull request merged**